zebra-chain/src/transparent.rs

```diff
@@ -105,26 +105,39 @@ pub enum Input {
     },
 }
 
-type Error = &'static str;
 impl Input {
     /// Get the value balance of this input.
     pub fn value_balance(
         &self,
         utxos: &HashMap<OutPoint, utxo::Utxo>,
-    ) -> Result<Amount<NonNegative>, Error> {
+    ) -> Result<Amount<NegativeAllowed>, InputError> {
```

Comment:
Minor: would it make sense to keep the `NonNegative` constraint? Also, @teor2345's #2511 might help here because they added a helper `Input::outpoint()` getter that returns an `Option<OutPoint>`.